### PR TITLE
Change package names without interior dir names

### PR DESCRIPTION
**Pull request by agocke**
The previous renaming properly renamed the package, but also renamed
the nested folder in the dotnet-runtime zip to Microsoft.NETCore.App.PGO,
which created problems with integration with the installer. This should
fix that by creating packages with a changed name without altering the contents.

### CODE REVIEW

**Comment by agocke** [comment]
@dotnet/runtime-infrastructure for review

**Review by jkoritzinsky — DISMISSED** [review]
LGTM

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <SkipBuild Condition="'$(RuntimeFlavor)' == 'Mono'">true</SkipBuild>
     <PlatformPackageType>RuntimePack</PlatformPackageType>
     <SharedFrameworkName>$(SharedFrameworkName).Crossgen2</SharedFrameworkName>
-    <OverridePackageId>$(SharedFrameworkName).$(RuntimeIdentifier)</OverridePackageId>
```
Comment (safern) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj:9]: DOn't we still need the `<OverridePackageId>$(SharedFrameworkName).$(RuntimeIdentifier)</OverridePackageId>` entry here?

Comment (agocke) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj:9]: Lemme check exactly what was going on here, I thought there was some unnecessary complexity, but maybe not.

Comment (agocke) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj:9]: Yup, you were right -- the package id override is still necessary. I confused the zip for the nuget package.

**Review by safern — APPROVED** [review]
LGTM

**Pull request merged**